docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,571 @@
+
+```{admonition} Note
+:class: note
+If you have a different use case or dataset you may need to process your data somehow (this is a very common task in machine learning)
+```
+## Tutorial
+
+### Define data and work directories
+
+We can start by creating an empty folder, I'll call mine `tutorial` and that's the name you'll see through the tutorial but feel free to use whatever you wish.
+
+We will display our results in our browser, so download the static folder from [here](https://github.com/jina-ai/jina/tree/master/jina/helloworld/chatbot/static), and paste it into your tutorial folder. This is only the CSS and HTML files to render our results. We will use a dataset in a `.csv` format. I'll use the [COVID](https://www.kaggle.com/xhlulu/covidqa) dataset from Kaggle. 
+You don't need to download this by hand, we'll do it later in our app.
+
+### Create a Flow
+
+The very first concept you'll see in Jina is a Flow. You can see {ref}`here <flow-cookbook>` a more formal introduction of what it is, but for now, think of the Flow as a manager in Jina, it takes care of the all the tasks that will run on your application and each Flow object will take care of one real-world task.
+
+To create a Flow you only need to import it from Jina. So open your favorite IDE, create an `app.py` file and let's start writing our code:
+
+``` python
+from jina import Flow
+flow = Flow()
+```
+
+But this is an empty Flow. Since we want to encode our data and then index it, we need to add elements to it. The only things we add to a Flow are Executors. We will talk about them more formally later, but think of them as the elements that will do all the data processing you want.
+
+### Add elements to a Flow
+
+To add elements to your Flow you just need to use the `.add()` method. You can add as many pods as you wish.
+
+``` python
+from jina import Flow
+flow = Flow().add().add()
+```
+
+And for our example, we need to add two `Executors`:
+
+1.  A transformer (to encode our data)
+2.  An indexer
+
+So add the following to our code:
+
+``` python
+from jina import Flow
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+    )
+```
+
+Right now we haven't defined `MyTransformer` or `MyIndexer`. Let's create some dummy Executors so we can try our app. These will not be our final Executors but just something basic to learn first.
+
+### Create dummy Executors
+
+Now we have a Flow with two Executors. Write the following in your code:
+
+``` python
+from jina import Executor, requests
+class MyTransformer(Executor):
+    @requests(on='/foo')
+    def foo(self, **kwargs):
+        print(f'foo is doing cool stuff: {kwargs}')
+class MyIndexer(Executor):
+    @requests(on='/bar')
+    def bar(self, **kwargs):
+        print(f'bar is doing cool stuff: {kwargs}')
```

Comment:
Add line breaks for readability - goes for all code snippets

Reply:
done